### PR TITLE
AKS Quickstart with helm chart integration

### PR DESCRIPTION
**Pull request by dciborow**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

* Deploy AKS cluster and configure with Helm Chart in a single ARM template

### CODE REVIEW

**Comment by azure-quickstarts** [comment]
@mlearned - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by azure-quickstarts** [comment]
@azure-gaming - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by azure-quickstarts** [comment]
@azure-gaming - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

File: quickstarts/microsoft.kubernetes/aks-helm/main.bicep [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+@maxValue(1023)
+param osDiskSizeGB int = 0
+
+@description('The number of nodes for the cluster.')
+@minValue(1)
+@maxValue(50)
+param agentCount int = 3
+
+@description('The size of the Virtual Machine.')
+param agentVMSize string = 'Standard_Ds_v3'
```
Comment (dciborow) [inline quickstarts/microsoft.kubernetes/aks-helm/main.bicep:21]:
```suggestion
param agentVMSize string = 'standard_a2'
```

**Comment by azure-quickstarts** [comment]
@azure-gaming - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by azure-quickstarts** [comment]
@azure-gaming - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

File: quickstarts/microsoft.kubernetes/aks-helm/metadata.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+  "$schema": "https://aka.ms/azure-quickstart-templates-metadata-schema#",
+  "type": "QuickStart",
+  "itemDisplayName": "Azure Container Service (AKS)",
+  "description": "Deploy a managed cluster with Azure Container Service (AKS)",
+  "summary": "Deploy a managed cluster with Azure Container Service (AKS)",
+  "githubUsername": "dciborow",
+  "docOwner": "azure-gaming",
```
Comment (dciborow) [inline quickstarts/microsoft.kubernetes/aks-helm/metadata.json:8]:
```suggestion
  "docOwner": "dciborow",
```

**Comment by azure-quickstarts** [comment]
@dciborow - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by azure-quickstarts** [comment]
@dciborow - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by azure-quickstarts** [comment]
@dciborow - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by azure-quickstarts** [comment]
@dciborow - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

File: quickstarts/microsoft.kubernetes/aks-helm/main.bicep [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  name: 'HelmScripts'
+  params: {
+    location                  : location
+    _artifactsLocation        : _artifactsLocation
+    _artifactsLocationSasToken: _artifactsLocationSasToken
+    clusterName               : clusterName
+    helmRepo                  : helmRepo
+    helmRepoURL               : helmRepoURL
+    helmApp                   : helmRepoApp
+    helmAppName               : helmRepoAppName
```
Comment (dciborow) [inline quickstarts/microsoft.kubernetes/aks-helm/main.bicep:72]:
```suggestion
    helmApp                   : helmApp
    helmAppName               : helmAppName
```

**Comment by azure-quickstarts** [comment]
@dciborow - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

File: quickstarts/microsoft.kubernetes/aks-helm/nested_template/helm.bicep [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+param helmApp string = 'azure-marketplace/wordpress'
+
+@description('Public Helm App Name')
+param helmAppName string = 'my-wordpress'
+
+var installScriptUri = uri(_artifactsLocation, 'scripts/helm.sh${_artifactsLocationSasToken}')
+
+var identityName = 'scratch${uniqueString(resourceGroup().id)}'
+var roleDefinitionId = resourceId('Microsoft.Authorization/roleDefinitions', 'b24988ac-6180-42a0-ab88-20f7382dd24c')
+var roleAssignmentName = guid(identityName, roleDefinitionId)
```
Comment (bmoore-msft) [inline quickstarts/microsoft.kubernetes/aks-helm/nested_template/helm.bicep:31]: this needs to contain the principal, role and scope as the seed - the identity name should be the resourceId of the MSI in the seed, e.g.

```bicep
guid(roleDefinitionId, managedIdentity.id, resourceGroup().id)
```

File: quickstarts/microsoft.kubernetes/aks-helm/nested_template/helm.bicep [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+resource managedIdentity 'Microsoft.ManagedIdentity/userAssignedIdentities@2018-11-30' = {
+  name: identityName
+  location: location
+}
+
+resource identityRoleAssignDeployment 'Microsoft.Authorization/roleAssignments@2020-04-01-preview' = {
+  scope: resourceGroup()
+  name: roleAssignmentName
+  properties: {
+    roleDefinitionId: '/subscriptions/${subscription().subscriptionId}/providers/Microsoft.Authorization/roleDefinitions/b24988ac-6180-42a0-ab88-20f7382dd24c'
```
Comment (bmoore-msft) [inline quickstarts/microsoft.kubernetes/aks-helm/nested_template/helm.bicep:42]: use the variable you have defined for this property value

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

**Comment by azure-quickstarts** [comment]
@dciborow - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by azure-quickstarts** [comment]
@dciborow - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by dciborow** [comment]
Could use some help with [TA-000023](https://github.com/Azure/template-analyzer/blob/main/docs/built-in-bpa-rules.md/#ta-000023-authorized-ip-ranges-should-be-defined-on-kubernetes-services) and [TA-000024](https://github.com/Azure/template-analyzer/blob/main/docs/built-in-bpa-rules.md/#ta-000024-role-based-access-control-rbac-should-be-used-on-kubernetes-services), the documentation that it links to only points to examples with az cli.

**Comment by azure-quickstarts** [comment]
@dciborow - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

File: quickstarts/microsoft.kubernetes/aks-helm/main.bicep [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+      adminUsername: linuxAdminUsername
+      ssh: {
+        publicKeys: [
+          {
+            keyData: sshRSAPublicKey
+          }
+        ]
+      }
+    }
+    kubernetesVersion: '1.14.0'
```
Comment (dciborow) [inline quickstarts/microsoft.kubernetes/aks-helm/main.bicep:76]:
```suggestion
    kubernetesVersion: '1.13.5'
```

**Comment by azure-quickstarts** [comment]
@dciborow - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

File: quickstarts/microsoft.kubernetes/aks-helm/modules/helm.bicep [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+      {
+        name: 'HELM_APP'
+        secureValue: helmApp
+      }
+      {
+        name: 'HELM_APP_NAME'
+        secureValue: helmAppName
+      }
+    ]
+    scriptContent: loadTextContent('../scripts/helm.sh'')
```
Comment (dciborow) [inline quickstarts/microsoft.kubernetes/aks-helm/modules/helm.bicep:82]:
```suggestion
    scriptContent: loadTextContent('../scripts/helm.sh')
```

**Comment by azure-quickstarts** [comment]
@dciborow - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by azure-quickstarts** [comment]
@dciborow - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

File: quickstarts/microsoft.kubernetes/aks-helm/azuredeploy.parameters.json [hunk 8]
```diff
@@ -0,0 +1,18 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "clusterName": {
```
Comment (dciborow) [inline quickstarts/microsoft.kubernetes/aks-helm/azuredeploy.parameters.json:5]:
```suggestion
    "aksName": {
```

**Comment by azure-quickstarts** [comment]
@dciborow - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by azure-quickstarts** [comment]
@dciborow - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by dciborow** [comment]
@bmoore-msft , i tried to address the BPA tag issues... but kept making things worse. Do these need to be addressed?

**Comment by azure-quickstarts** [comment]
@dciborow - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Pull request merged**